var/ramble/repos/builtin/modifiers/gcp-metadata/modifier.py

```diff
@@ -24,23 +24,62 @@ class GcpMetadata(BasicModifier):
     mode('standard', description='Standard execution mode')
     default_mode('standard')
 
-    variable_modification('log', '{experiment_run_dir}/gcp-metadata.log', method='set', modes=['standard'])
-    archive_pattern('{log}')
+    executable_modifier('gcp_metadata_exec')
 
-    register_builtin('gcp_metadata_exec')
+    def gcp_metadata_exec(self, executable_name, executable, app_inst=None):
```

Comment:
One odd gotcha with this is you're taking every executable in the experiment and prepending this to it. So, you'll be `rm`-ing and re-creating the same files many times. You might want to add and instance tracker (which is a bit gross) to make sure you only do it once.

Something like:

```
def gcp_metadata_exec(....):
  if not hasattr(self, '_apply_gcp_metadata_exec'):
    self._apply_gcp_metadata_exec = False
    ....
```

Reply:
Great catch! I meant to add that, but got carried away in the details. Thanks, will ad it on